File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.*;

import java.util.Map;
import java.util.regex.Pattern;

public class TransformerFunctionTest extends TransformerFunction {
    public TransformerFunctionTest() {
        super(FunctionDescription.of(
                Map.of(
                        "pattern", ArgumentType.of(ArgType.String).position(0)
                )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var str = context.getString(null);
        if (str == null) {
            return false;
        }
        var patternString = context.getString("pattern");
        if (patternString == null) {
            return false;
        }
        return Pattern.compile(patternString).matcher(str).find();
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionTest.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionTest extends TransformerFunction {
  constructor() {
    super({
      arguments: {
        pattern: { type: ArgType.String, position: 0 },
      },
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    if (str == null) {
      return false;
    }
    let patternString = await context.getString("pattern");
    if (patternString == null) {
      return false;
    }
    let flags = "";
    if (patternString.startsWith("(?i)")) {
      flags = "i";
      patternString = patternString.substring(4);
    }
    if (patternString.startsWith("(?m)")) {
      flags += "m";
      patternString = patternString.substring(4);
    }
    return new RegExp(patternString, flags).test(str);
  }
}

export default TransformerFunctionTest;
